### PR TITLE
tikv-ctl: print region size

### PR DESCRIPTION
**Pull request by choleraehyq**
@zhangjinpeng1987 PTAL

- [x] Prettify output format

### CODE REVIEW

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -154,13 +161,22 @@ fn main() {
             let skip_tombstone = matches.is_present("skip-tombstone");
             match matches.value_of("region") {
                 Some(id) => {
-                    dump_region_info(&db, String::from(id).parse().unwrap(), skip_tombstone)
+                    dump_region_info(&db, String::from(id).parse().unwrap(), skip_tombstone);
```
Comment (andelf) [inline src/bin/tikv-ctl.rs]: id.parse()

**Comment by choleraehyq** [comment]
@zhangjinpeng1987 PTAL

File: src/bin/tikv-ctl.rs (hunk 1 above)
Comment (choleraehyq) [inline src/bin/tikv-ctl.rs]: Done.

File: src/bin/tikv-ctl.rs [hunk 2]
```diff
@@ -355,21 +371,111 @@ fn dump_region_info(db: &DB, region_id: u64, skip_tombstone: bool) {
...
+}
+
+fn dump_all_region_size(db: &DB) {
+    let region_ids = get_all_region_id(db);
+    let mut v: Vec<(u64, u64)> = Vec::new();
+    let mut total_size = 0;
+    let mut region_number = 0;
+    for region_id in region_ids {
+        let size = get_region_size(db, region_id);
+        v.push((region_id, size));
```
Comment (andelf) [inline src/bin/tikv-ctl.rs]: How about make it (size, region) tuple. Avoid reimplementing cmp?

File: src/bin/tikv-ctl.rs [hunk 3]
```diff
@@ -355,21 +371,111 @@ fn dump_region_info(db: &DB, region_id: u64, skip_tombstone: bool) {
...
+        total_size += size;
+        region_number += 1;
```
Comment (andelf) [inline src/bin/tikv-ctl.rs]: This code is more like C instead of Rust.
Try use iterator, Vec member functions.

File: src/bin/tikv-ctl.rs [hunk 4]
```diff
@@ -355,21 +370,91 @@ fn dump_region_info(db: &DB, region_id: u64, skip_tombstone: bool) {
...
+fn dump_all_region_info(db: &DB, skip_tombstone: bool) {
+    let region_ids = get_all_region_id(db);
+    for region_id in region_ids {
+        dump_region_info(db, region_id, skip_tombstone);
+    }
+}
+
+fn dump_all_region_size(db: &DB) {
+    let region_ids = get_all_region_id(db);
+    let (total_size, region_number) = region_ids.iter().fold((0, 0), |acc, &region_id| {
```
Comment (andelf) [inline src/bin/tikv-ctl.rs]: region_number = v.len()
You can use map + sum to calculate total_size

**Comment by choleraehyq** [comment]
@andelf 
Learn a lot from your reviews, thanks. All done.

File: src/bin/tikv-ctl.rs [hunk 5]
```diff
@@ -353,21 +368,91 @@ fn dump_region_info(db: &DB, region_id: u64, skip_tombstone: bool) {
     println!("apply state: {:?}", apply_state);
 }
 
-fn dump_all_region_info(db: DB, skip_tombstone: bool) {
+fn convert_gbmb(mut bytes: u64) -> String {
+    let mb = if bytes % (1024 * 1024 * 1024) == 0 {
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: use const defined GB, MB

File: src/bin/tikv-ctl.rs [hunk 6]
```diff
@@ -353,21 +368,91 @@ fn dump_region_info(db: &DB, region_id: u64, skip_tombstone: bool) {
...
+    v.reverse();
+    println!("total region number: {}", region_number);
+    println!("total region size: {}", convert_gbmb(total_size));
+    for (size, id) in v {
+        println!("region_id: {}", id);
+        println!("region size: {}", convert_gbmb(size));
+    }
+}
+
+fn get_all_region_id(db: &DB) -> Vec<u64> {
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: get_all_region_ids

File: src/bin/tikv-ctl.rs [hunk 7]
```diff
@@ -353,21 +368,91 @@ fn dump_region_info(db: &DB, region_id: u64, skip_tombstone: bool) {
...
     let start_key = keys::REGION_META_MIN_KEY;
     let end_key = keys::REGION_META_MAX_KEY;
+    let mut region_ids: Vec<u64> = Vec::new();
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: seem you can use `vec!` and remove the type declaration here.

File: src/bin/tikv-ctl.rs (hunk 5 above)
Comment (andelf) [inline src/bin/tikv-ctl.rs]: Size humanrize is very common in tikv code base.
For future improvements, https://github.com/Thomasdezeeuw/human-size-rs/blob/master/src/lib.rs is a good implementation.

**Comment by choleraehyq** [comment]
@zhangjinpeng1987 Merge?

**Pull request merged**